backend_typescript/server/gameApi.ts

```diff
@@ -0,0 +1,326 @@
+    }
+    req.game = userGame.result;
+    next();
+});
+
+
+gameRouter.use('/games/:game_id/place', async (req, res: Response, next: NextFunction) => {
+    const user = req.user as User;
+    const game = req.game as Game;
+    const gameShipManager = game.getGameShipManager(user.id);
+    if (!(gameShipManager instanceof ShipManager)) {
+        return res.status(400).json({msg: gameShipManager});
+    }
+    const shipManager = gameShipManager as ShipManager;
+    req.shipManager = shipManager;
+    next();
+});
+
+
+// Get user profile page
+gameRouter.get("/profile", async (req, res: Response) => {
+    try {
+        const user = req.user as User;
+        const userData = {
+            username: user.username,
+            nickname: user.nickname,
+            user_score: {
+                wins: user.wins,
+                loses: user.loses
+            },
+            imageName: user.imgName,
+            imgId: user.imgId
+        }
+        res.status(200).json(userData);
+    } catch (e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+
+// Update user profile page
+gameRouter.put('/profile', async (req, res: Response, next: NextFunction) => { 
+    const { nickname, imgId }: ProfileUpdateBody = req.body;
+    const user = req.user as User;
+    if (!nickname) {
+        return res.status(400).json({msg: 'Nickname must be specified'});
+    }
+    
+    try {
+        await user.updateProfile(imgId, nickname);
+        res.status(200).json({ msg: 'Updated user' });
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+
+// Image Upload & Retrieve
+const imageUpload = multer({
+    storage: multer.diskStorage(
+        {
+            destination: function (req, file, cb) {
+                cb(null, 'images/');
+            },
+            filename: function (req, file, cb) {
+                cb(
+                    null,
+                    new Date().valueOf() + 
+                    '_' +
+                    file.originalname
+                );
+            }
+        }
+    ), 
+});
+// Image Upload Routes
+gameRouter.post('/image',
+    imageUpload.single('image'),
+    async (req, res) => {
+        try {
+            const result = await db.insertImage(req.file.filename, req.file.path, req.file.mimetype, req.file.size);
+            res.status(200).json(result);
+        } catch (e) {
+            console.log(e);
+            res.status(500).json({ msg: 'Server error' });
+        }
+});
+// Image Get Routes
+gameRouter.get('/image/:filename', (req, res) => {
+    const { filename } = req.params;
+    const dirname = path.resolve();
+    const fullfilepath = path.join(dirname, 'images/' + filename);
+    return res.sendFile(fullfilepath);
+});
+
+
+// Create a new game
+gameRouter.post('/games', async (req, res: Response) => {
+    const {opponent, dimension}: GameBody = req.body;
+    const user = req.user as User;
+
+    if (dimension !== 10 && dimension !== 20) {
```

Comment:
it is better to move these two verifications inside the function `gameManager.createGame`